controllers/iamauthpolicy_controller.go

```diff
@@ -249,13 +275,50 @@ func (c IAMAuthPolicyController) updatePolicyCondition(ctx context.Context, k8sP
 		status = metav1.ConditionFalse
 	}
 	cnd := metav1.Condition{
-		Type:               string(gwv1alpha2.PolicyConditionAccepted),
-		Status:             status,
-		ObservedGeneration: k8sPolicy.Generation,
-		LastTransitionTime: metav1.Now(),
-		Reason:             string(reason),
+		Type:   string(gwv1alpha2.PolicyConditionAccepted),
+		Status: status,
+		Reason: string(reason),
 	}
 	meta.SetStatusCondition(&k8sPolicy.Status.Conditions, cnd)
 	err := c.client.Status().Update(ctx, k8sPolicy)
 	return err
 }
+
+func iamAuthPolicyMapFunc(c client.Client, log gwlog.Logger) handler.MapFunc {
+	return func(obj client.Object) []ctrl.Request {
+		requests := []ctrl.Request{}
+		policies := &anv1alpha1.IAMAuthPolicyList{}
+		err := c.List(context.Background(), policies, &client.ListOptions{Namespace: obj.GetNamespace()})
+		if err != nil {
+			log.Error(err)
+			return requests
+		}
+		for _, policy := range policies.Items {
+			if obj.GetName() == string(policy.Spec.TargetRef.Name) {
+				requests = append(requests, ctrl.Request{NamespacedName: policy.GetNamespacedName()})
+			}
+		}
+		return requests
+	}
+}
+
+// TODO: move into services package after Erik's target group renaming
+func ignoreTargetRefNotFound(err error) error {
+	if services.IsNotFoundError(err) {
+		return nil
+	}
+	return err
+}
+
+func (c *IAMAuthPolicyController) handleTargetRefNotFound(ctx context.Context, k8sPolicy *anv1alpha1.IAMAuthPolicy, err error) error {
+	if services.IsNotFoundError(err) {
+		err = c.updatePolicyCondition(ctx, k8sPolicy, gwv1alpha2.PolicyReasonTargetNotFound)
+	}
+	return err
+}
+
+func (c *IAMAuthPolicyController) updateLatticeAnnotaion(ctx context.Context, k8sPolicy *anv1alpha1.IAMAuthPolicy, resId string) error {
+	k8sPolicy.Annotations["application-networking.k8s.aws/resourceId"] = resId
```

Comment:
Define a constant for Annotations key?

https://github.com/aws/aws-application-networking-k8s/blob/a2d5b973b3516f11784d1a764830e8de2eadf4a7/pkg/apis/applicationnetworking/v1alpha1/accesslogpolicy_types.go#L13

Reply:
it's not used anywhere else